src/resources/views/crud/fields/repeatable.blade.php

```diff
@@ -34,41 +27,76 @@
+                @foreach($field['fields'] as $subfield)
+                    @php
+                        $subfield = $crud->makeSureFieldHasNecessaryAttributes($subfield);
+                        $fieldViewNamespace = $subfield['view_namespace'] ?? 'crud::fields';
+                        $fieldViewPath = $fieldViewNamespace.'.'.$subfield['type'];
+                        if(!is_array($subfield['name'])) {
+                            if(isset($row[$subfield['name']])) {
+                                $subfield['value'] = $row[$subfield['name']];
+                            }
+                        }else{
+                            
+                            $subfield['value'] = $row;
+                        }                       
+                    @endphp
+
+                    @include($fieldViewPath, ['field' => $subfield])
+                @endforeach
+
+                @endif
+            </div>
         @endforeach
-
-      @endif
+    @endif
     </div>
-    @endpush
-
-  </div>
+</div>
 
-
-  <button type="button" class="btn btn-outline-primary btn-sm ml-1 add-repeatable-element-button">+ {{ $field['new_item_label'] ?? trans('backpack::crud.new_item') }}</button>
+{{-- HINT --}}
+@if (isset($field['hint']))
+    <p class="help-block text-muted text-sm">{!! $field['hint'] !!}</p>
+@endif
+<button type="button" class="btn btn-outline-primary btn-sm ml-1 add-repeatable-element-button">+ {{ $field['new_item_label'] ?? trans('backpack::crud.new_item') }}</button>
 
 @include('crud::fields.inc.wrapper_end')
-
+@push('before_scripts')
+<div class="col-md-12 well repeatable-element row m-1 p-2" data-repeatable-identifier="{{ $field['name'] }}">
+  @if (isset($field['fields']) && is_array($field['fields']) && count($field['fields']))
+    <div class="controls">
+        <button type="button" class="close delete-element"><span aria-hidden="true">×</span></button>
+        <button type="button" class="close move-element-up">
+            <svg viewBox="0 0 64 80"><path d="M46.8,36.7c-4.3-4.3-8.7-8.7-13-13c-1-1-2.6-1-3.5,0c-4.3,4.3-8.7,8.7-13,13c-2.3,2.3,1.3,5.8,3.5,3.5c4.3-4.3,8.7-8.7,13-13c-1.2,0-2.4,0-3.5,0c4.3,4.3,8.7,8.7,13,13C45.5,42.5,49,39,46.8,36.7L46.8,36.7z"/></svg>
+        </button>
+        <button type="button" class="close move-element-down">
+            <svg viewBox="0 0 64 80"><path d="M17.2,30.3c4.3,4.3,8.7,8.7,13,13c1,1,2.6,1,3.5,0c4.3-4.3,8.7-8.7,13-13c2.3-2.3-1.3-5.8-3.5-3.5c-4.3,4.3-8.7,8.7-13,13c1.2,0,2.4,0,3.5,0c-4.3-4.3-8.7-8.7-13-13C18.5,24.5,15,28,17.2,30.3L17.2,30.3z"/></svg>
+        </button>
+    </div>
+    @foreach($field['fields'] as $subfield)
+      @php
+          $subfield = $crud->makeSureFieldHasNecessaryAttributes($subfield);
+          $fieldViewNamespace = $subfield['view_namespace'] ?? 'crud::fields';
+          $fieldViewPath = $fieldViewNamespace.'.'.$subfield['type'];
+      @endphp
+      @include($fieldViewPath, ['field' => $subfield])
+    @endforeach
+  @endif
+</div>
```

Comment:
@pxpm I see we iterate through the fields twice now, in the same blade file. And output the same HTML twice - once inside the form, once outside it (pushed to `before_scripts`). 

I'll take that as a red flag 🚩  Why is this needed? Previously we were just doing that once.

Reply:
Kind of , in the first iteration we use the field value `@foreach ($field['value'] as $key => $row)` to create the appropriate field for each value stored in the database. For each row in the value we create the fields in the page. If there is no value (create operation for example) that iteration does not happen!

The second iteration is `the default state row`, we iterate the fields in repeatable creating the `empty state form fields` that we push to the DOM when needed. 

When there is value (1st iteration) we set the field value accordingly like when they are defined as a regular crud field in the update operation.

Reply:
Ok, so that means... 
- the first one is the actual HTML that the user sees;
- the second one is a hidden HTML that we use to add new rows;

Why do we need this code duplication now, and didn't need it before? Because we now populate default repeatable values using PHP instead of JS? Why do we do that anyway?

Reply:
We do that explicitly to avoid **initing empty fields** that technically **are not empty but we set the value later with JS**. 
This makes fields that depend on field value to setup stuff work inside repeatable, I am looking at you images and uploads ... probably datetimepicker too. Ahh and I think toggle is also in this list since it works in this branch and not in the master. 

This way fields are added to the page like any regular crud form field, we just setup the rows, selectors, array names etc after with JS, but on rendering, they are regular crud fields, with value and everything they deserve.

Reply:
Ok got it. Last time we couldn't output with default directly, now we can. But it's very not-DRY to have the same code twice in the same file.

Can we create a function inside this file... say `outputRepeatableElements()` that will do the output, and we can call that twice? It would also need to check the function doesn't already exist - in case `repeatable` is used twice in the same form.

That would clean up the blade file a bit, don't you think?